WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/UsernameChangerFullScreenDialogFragment.java

```diff
@@ -0,0 +1,322 @@
+        ActivityUtils.hideKeyboard(getActivity());
+
+        Bundle result = new Bundle();
+        result.putString(RESULT_USERNAME, mUsernamesAdapter.mItems.get(mUsernamesAdapter.getSelectedItem()));
+        controller.confirm(result);
+
+        return true;
+    }
+
+    @Override
+    public boolean onDismissClicked(FullScreenDialogController controller) {
+        ActivityUtils.hideKeyboard(getActivity());
+
+        if (hasUsernameChanged()) {
+            showDismissDialog();
+        } else {
+            controller.dismiss();
+        }
+
+        return true;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean(KEY_SHOULD_WATCH_TEXT, false);
+        outState.putString(KEY_USERNAME_SELECTED, mUsernameSelected);
+        outState.putInt(KEY_USERNAME_SELECTED_INDEX, mUsernameSelectedIndex);
+        outState.putStringArrayList(KEY_USERNAME_SUGGESTIONS, new ArrayList<>(mUsernamesAdapter.mItems));
+    }
+
+    @Override
+    public void onUsernameSelected(String username) {
+        mHeaderView.setText(getHeaderText(username, mDisplayName));
+        mUsernameSelected = username;
+        mUsernameSelectedIndex = mUsernamesAdapter.getSelectedItem();
+    }
+
+    protected Spanned getHeaderText(String username, String display) {
+        return Html.fromHtml(
+                String.format(
+                        getString(R.string.username_changer_header),
+                        "<b>",
+                        username,
+                        "</b>",
+                        "<b>",
+                        display,
+                        "</b>"
+                )
+        );
+    }
+
+    protected String getUsernameOrSelected() {
+        return TextUtils.isEmpty(mUsernameSelected) ? mUsername : mUsernameSelected;
+    }
+
+    protected void getUsernameSuggestions(String usernameQuery) {
+        showProgress(true);
+
+        FetchUsernameSuggestionsPayload payload = new FetchUsernameSuggestionsPayload(usernameQuery);
+        mDispatcher.dispatch(AccountActionBuilder.newFetchUsernameSuggestionsAction(payload));
+    }
+
+    private String getUsernameQueryFromDisplayName() {
+        return mDisplayName.replace(" ", "").toLowerCase();
+    }
+
+    protected boolean hasUsernameChanged() {
+        return !TextUtils.equals(mUsername, mUsernameSelected);
+    }
+
+    private void populateUsernameSuggestions(List<String> suggestions) {
+        String usernameOrSelected = getUsernameOrSelected();
+
+        List<String> suggestionList = new ArrayList<>();
+        suggestionList.add(usernameOrSelected);
+
+        for (String suggestion : suggestions) {
+            if (!TextUtils.equals(suggestion, usernameOrSelected)) {
+                suggestionList.add(suggestion);
+            }
+        }
+
+        mUsernamesAdapter = new UsernameChangerRecyclerViewAdapter(getActivity(), suggestionList);
```

Comment:
Perhaps here's an opportunity to reuse `setUsernameSuggestions()` after setting `mUsernameSelectedIndex = 0`?

Reply:
I updated the method in 71818d2200bf79922b4c4bcfbfcf76cf6cb1fb69.